classes/core/PKPHtmlSanitizer.php

```diff
@@ -0,0 +1,159 @@
+        )->keys();
+
+        $this->buildSanitizerConfig(
+            $this->generateAllowedTagToAttributeMap(
+                Config::getVar('security', $allowable, null) ?? $allowable
+            )
+        );
+    }
+
+    /**
+     * Set a pre defined instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function setSanitizerConfig(HtmlSanitizerConfig $htmlSanitizerConfig): self
+    {
+        $this->htmlSanitizerConfig = $htmlSanitizerConfig;
+
+        return $this;
+    }
+
+    /**
+     * Get the instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function getSanitizerConfig(): HtmlSanitizerConfig
+    {
+        return $this->htmlSanitizerConfig;
+    }
+
+    /**
+     * Sanitize the given html string
+     */
+    public function sanitize(string $html): string
+    {   
+        return (new HtmlSanitizer($this->htmlSanitizerConfig))->sanitize(
+            // Here we are removing any html tags that should not be handled by sanitizer
+            strip_tags($html, $this->getSanitizableTags()->toArray())
+        );
+    }
+
+    /**
+     * Build up the \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig instance
+     */
+    protected function buildSanitizerConfig(Collection $allowedTagToAttributeMap): void
+    {
+        $this->htmlSanitizerConfig = (new HtmlSanitizerConfig())
+            ->allowLinkSchemes(['https', 'http', 'mailto'])
+            ->allowMediaSchemes(['https', 'http']);
+        
+        if ($allowedTagToAttributeMap->count()) {
+            $allowedTagToAttributeMap->each(
+                fn (array $attributes, string $tag) => $this->htmlSanitizerConfig = $this->htmlSanitizerConfig->allowElement($tag, $attributes)
+            );
+        }
+
+        $this->getNonAllowedHtmlTags()->each(
+            fn (string $tag) => $this->htmlSanitizerConfig = $this->htmlSanitizerConfig->blockElement($tag)
+        );
+    }
+
+    /**
+     * Get the collection of non allowed tags for the given configuration
+     */
+    protected function getNonAllowedHtmlTags(): Collection
+    {
+        return $this->w3cValidElements
+            ->diff(collect($this->htmlSanitizerConfig->getAllowedElements())->keys());
+    }
+
+    /**
+     * Get the collection of tags that will not be stripped/removed by php's "strip_tags" function
+     * but rather should be handled by the sanitizer class
+     */
+    protected function getSanitizableTags(): Collection
+    {
+        return $this->w3cValidElements
+            ->merge(collect($this->htmlSanitizerConfig->getAllowedElements())->keys())
+            ->merge(collect([ // list of dangerous tags that should be only handled by sanitization library
+                'script',
+            ]))
+            ->unique();
+    }
+
+    /**
+     * Generate the collection of allowed tags to allowed attributes map as key/value[array] structure
```

Comment:
Needs self-documentation on parameter and return; they aren't clear to me at a glance over the function definition.

Reply:
Right, it's a bit complex logic and hard to understand what it's doing , DocBlock added .